File: tournaments/isInformationConsistent/isInformationConsistent.js
function isInformationConsistent(evidences) {
    for (let j = 0; j in evidences[0]; ++j) {
        let r = 0;
        for (let i = 0; i in evidences; ++i) {
            if (1 === evidences[i][j]) {
                if (!r) r = 1;
                else if (!~r) return false;
            } else if (!~evidences[i][j]) {
                if (!r) r = -1;
                else if (1 === r) return false;
            }
        }
    }

    return true;
}

File: tournaments/isInformationConsistent/isInformationConsistent.pl
#!/usr/bin/perl
use strict;
use warnings FATAL => 'all';

sub isInformationConsistent {
    my ($evidences) = @_;

    my $rows = @$evidences;
    my $cols = @{@$evidences[0]};

    for (my $col = 0; $cols > $col; ++$col) {
        my $c = 0;
        for (my $row = 0; $rows > $row; ++$row) {
            if (!@{@$evidences[$row]}[$col]) {
                next;
            } elsif (1 == @{@$evidences[$row]}[$col]) {
                if (!$c) {
                    $c = 1;
                } elsif (-1 == $c) {
                    return 0;
                }
            } else {
                if (!$c) {
                    $c = -1;
                } elsif (1 == $c) {
                    return 0;
                }
            }
        }
    }

    return 1;
}
